overlord/snapstate/backend/snapdata.go

```diff
@@ -113,6 +121,21 @@ func removeDirs(dirs []string) error {
 	return nil
 }
 
+// snapDataDirs returns the list of base data directories for the given snap name
+func snapBaseDataDirs(name string, opts *dirs.SnapDirOptions) ([]string, error) {
+	// collect the directories, homes first
+	found, err := filepath.Glob(snap.BaseDataHomeDir(name, opts))
```

Comment:
Great catch, thank you! I updated to support non-standard homedirs and also updated the spread test to check for this.
**Note:**
It looks like `system.homedirs` are not supported by other removal methods for common data and snap revision data, this means that the revision data of all revision will just lay around forever until the snap is removed.
Most importantly it doesn't seem supported for [copy operation ](https://github.com/snapcore/snapd/blob/15c522cc930029e9a9432d4625a82b8f0156786a/overlord/snapstate/backend/snapdata.go#L159)which could cause unexpected data loss for users with non-standard home dirs after a refresh, this looks like a bug.

Reply:
Thanks for looking into this, please add a jira card about the missing copy support that looks indeed like a nasty bug (and it seems not that many snaps use revisioned data or it would have been on our radar earlier).